Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift

```diff
@@ -91,12 +87,64 @@ extension RangeReplaceableCollection {
         return remove(at: index)
     }
 
-    #if canImport(Foundation)
     /// SwifterSwift: Remove a random value from the collection.
     @discardableResult public mutating func removeRandomElement() -> Element? {
         guard let randomIndex = indices.randomElement() else { return nil }
         return remove(at: randomIndex)
     }
-    #endif
+
+    /// SwifterSwift: Keep elements of Array while condition is true.
+    ///
+    ///        [0, 2, 4, 7].keep(while: { $0 % 2 == 0 }) -> [0, 2, 4]
+    ///
+    /// - Parameter condition: condition to evaluate each element against.
+    /// - Returns: self after applying provided condition.
+    /// - Throws: provided condition exception.
+    @discardableResult
+    public mutating func keep(while condition: (Element) throws -> Bool) rethrows -> Self {
```

Comment:
This can be simplified with:
```Swift
public mutating func keep(while condition: (Element) throws -> Bool) rethrows -> Self {
    if let lastIndex = try firstIndex(where: { try !condition($0) }) {
        removeSubrange(lastIndex...)
    }
    return self
}
```
I also don't see why a `mutating` function should return itself, but no need to break backwards compatibility.